feathr_project/registry/registry/models.py

```diff
@@ -0,0 +1,707 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Optional, Union
+from uuid import UUID
+import json
+import re
+
+
+def to_snake(d):
+    if isinstance(d, str):
+        return re.sub('([A-Z]\w+$)', '_\\1', d).lower()
+    if isinstance(d, list):
+        return [to_snake(i) if isinstance(i, (dict, list)) else i for i in d]
+    return {to_snake(a): to_snake(b) if isinstance(b, (dict, list)) else b for a, b in d.items()}
+
+
+def to_type(value, type):
```

Comment:
Add description for this method. Also we should add hint for the input/output types.

Reply:
The type hint is omitted purposely, this function is for dynamically type casting, the input value could be anything and the output value is in the type specified by `type` parameter.
This is not something can be handled by Python typing mechanism.